### PR TITLE
DEVDOCS-3021: Update Channel-webhooks

### PR DESCRIPTION
**Pull request by traci707**
# [DEVDOCS-3021](https://jira.bigcommerce.com/browse/DEVDOCS-3021)

## What changed?
* thing_that_changed

### CODE REVIEW

File: docs/api-docs/channels/channel-webhooks.md [hunk 1]
```diff
@@ -25,6 +25,17 @@ Updates to the following fields trigger a `store/channel/updated` event:
 * `is_enabled` (to be deprecated)
 * `config_meta`
 
+## Channel-aware webhook events
+
+| Name                                                   | Description                                                    |
+|-----------------------------------------------------   |----------------------------------------------------------      |
+| store/channel/{channel_id}/product/assigned            | New product assigned to channel (or new listing creating)      |
```
Comment (kristinapototska) [inline docs/api-docs/channels/channel-webhooks.md:32]: Product don't need to be new. Those webhooks would trigger when any product is assigned or unassigned to / from channel (via both catalog and listings API).

**Review by kristinapototska — CHANGES_REQUESTED** [review]

**Comment by slsriehl** [comment]
@traci707 @markcmurphy I added more context for MSF and table columns with links to the endpoints that trigger these webhooks. Do we have a ticket to add these new MSF channel-specific webhooks to our list of schemas?

File: docs/api-docs/channels/channel-webhooks.md [hunk 2]
```diff
@@ -25,58 +24,65 @@ Updates to the following fields trigger a `store/channel/updated` event:
...
+| store/channel/{channel_id}/product/unassigned          | Fires when a product is removed from the specified channel | [Delete product channel assignments](/api-reference/store-management/catalog/products-channel-assignments/deleteproductchannelassignments) |
+| store/channel/{channel_id}/category/product/assigned   | Fires when a product is assigned to a category in the specified channel's category tree | [Create product category assignments](/api-reference/store-management/catalog/products-category-assignments/deleteproductscategoryassignments) |
+| store/channel/{channel_id}/category/product/unassigned | Fires when a product is removed from a category in the specified channel's category tree | [Delete product category assignments](/api-reference/store-management/catalog/products-category-assignments/createproductscategoryassignments) |
+| store/channel/{channel_id}/settings/profile/updated    | Fires when the store profile settings that apply to the specified channel are updated | [Update store profile settings](/api-reference/store-management/settings/store-profile/putstoreprofilesettings) |
```
Comment (slsriehl) [inline docs/api-docs/channels/channel-webhooks.md:37]: @kristinapototska 
Does the `store/channel/{channel_id}/settings/profile/updated` webhook fire only when channel-specific overrides are updated, or also when global store settings are updated for which channel-specific overrides do not exist?

cc @traci707 @markcmurphy

**Comment by traci707** [comment]
@slsriehl Are you talking about [DEVDOCS-3021[(https://jira.bigcommerce.com/browse/DEVDOCS-3021)

**Comment by kristinapototska** [comment]
@slsriehl given webhooks works as an overlay to fire every time store profile settings are updated in a way that affects channel you subscribed for. 
If you are subscribed to store profile changes per e.g. channel 5 and:
1. Given channel has store profile settings overridden – webhook would fire when given settings are updated on channel level.
2. Given channel is inheriting store profile settings from global level – webhook would fire when store profile settings are updated on global level which also has impact on given channel.


If there are 2 channels that are inheriting store profile settings from global level, when settings are changed on global level we would fire webhooks for each of those 2 channels.
This is going to work exactly the same for all the webhooks for settings where we have both global and channel level overrides.

**Comment by slsriehl** [comment]
@traci707 I don't think so. What I mean to ask is whether the sidebar list of webhook subscription points on the `/api-reference` page is slated to be updated. If 3021 also includes those changes as part of the work, great. If not, we need to find/create a ticket for that.

**Comment by traci707** [comment]
@slsriehl I will create a separate ticket to update the side-bar links. I would like to get this PR merged.

**Review by slsriehl — APPROVED** [review]
@traci707 this is GTG from my POV. Please push back against my changes if you don't like them or they don't seem right.

**Pull request merged**